cmd/carapace/cmd/lazyinit/nushell.go

```diff
@@ -5,13 +5,26 @@ import "fmt"
 func Nushell(completers []string) string {
 	snippet := `%v%v
 
-let carapace_completer = {|spans| 
-  carapace $spans.0 nushell $spans | from json
+let carapace_completer = {|spans|
+  # if the current command is an alias, get it's expansion
+  let expanded_alias = (scope aliases | where name == $spans.0 | get -i 0 | get -i expansion)
+
+  # overwrite
+  let spans = (if $expanded_alias != null  {
+    # put the first word of the expanded alias first in the span
+    $spans | skip 1 | prepend ($expanded_alias | split row " ")
+  } else {
+    $spans
+  })
+
+  carapace $spans.0 nushell $spans
+  | from json
+  | if ($in | default [] | where value =~ '^-.*ERR$' | is-empty) { $in } else { null }
```

Comment:
This hides the [error messages](https://carapace.sh/#give-feedback-with-messages)? Those are intentional.

Reply:
Just seeing [this](https://www.nushell.sh/cookbook/external_completers.html#err-unknown-shorthand-flag-using-carapace).
There is `$env.CARAPACE_LENIENT = 1` if one wants to allow unknown flags.

Flag parsing is strict for a reason though so this will have some implications.
For unknown flags it is not clear if the next argument is positional or belonging to the flag.
So all unknown flags get treated pretty much like boolean/optarg flags.

Reply:
Sounds reasonable to enable error pass through. I get to pc later. Ill edit the script and push the reworked init.

Reply:
Removed the last line and checked the behavior in my setup. Works flawless and errors get shown.